최창익/app/src/main/java/com/mash_up/mvvmstudy/view/main/MainViewModel.kt

```diff
@@ -0,0 +1,37 @@
+package com.mash_up.mvvmstudy.view.main
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.mash_up.mvvmstudy.repository.model.Repository
+import com.mash_up.mvvmstudy.repository.MainRepository
+
+class MainViewModel : ViewModel() {
+    private val mainRepository = MainRepository()
+
+    private val _isLoading: MutableLiveData<Boolean> = MutableLiveData()
+    val isLoading: LiveData<Boolean>
+        get() = _isLoading
+
+    private val _repositories: MutableLiveData<List<Repository>> = MutableLiveData()
+    val repositories: LiveData<List<Repository>>
+        get() = _repositories
+
+    private val _networkErrorState: MutableLiveData<Unit> = MutableLiveData()
+    val networkErrorState: LiveData<Unit>
+        get() = _networkErrorState
+    var networkErrorInfo: String = ""
```

Comment:
혹시 networkErrorInfo를 따로 두는 이유가 있을까요? 
```suggestion
    private val _networkErrorState: MutableLiveData<String> = MutableLiveData()
    val networkErrorState: LiveData<String>
        get() = _networkErrorState
```